crates/teloxide-core/src/types/message.rs

```diff
@@ -1087,76 +1095,130 @@ mod getters {
+        pub fn super_group_chat_created(&self) -> Option<&MessageSupergroupChatCreated> {
             match &self.kind {
-                SupergroupChatCreated(MessageSupergroupChatCreated { supergroup_chat_created }) => {
-                    Some(*supergroup_chat_created)
+                SupergroupChatCreated(message_supergroup_chat_created) => {
+                    Some(message_supergroup_chat_created)
                 }
                 _ => None,
             }
         }
 
+        /// Returns `true` if the incoming [`Message`] contains the
+        /// `channel_chat_created` Service message.
+        ///
+        /// [More on this](https://core.telegram.org/bots/api#message)
         #[must_use]
-        pub fn channel_chat_created(&self) -> Option<True> {
+        pub fn is_channel_chat_created(&self) -> bool {
             match &self.kind {
-                ChannelChatCreated(MessageChannelChatCreated { channel_chat_created }) => {
-                    Some(*channel_chat_created)
+                ChannelChatCreated(..) => true,
+                _ => false,
+            }
+        }
+
+        #[must_use]
+        pub fn channel_chat_created(&self) -> Option<&MessageChannelChatCreated> {
+            match &self.kind {
+                ChannelChatCreated(message_channel_chat_created) => {
+                    Some(message_channel_chat_created)
                 }
                 _ => None,
             }
         }
 
         #[must_use]
-        pub fn chat_migration(&self) -> Option<ChatMigration> {
+        pub fn message_auto_delete_timer_changed(
+            &self,
+        ) -> Option<&types::MessageAutoDeleteTimerChanged> {
+            match &self.kind {
+                MessageAutoDeleteTimerChanged(MessageMessageAutoDeleteTimerChanged {
+                    message_auto_delete_timer_changed,
+                }) => Some(message_auto_delete_timer_changed),
+                _ => None,
+            }
+        }
+
+        #[must_use]
+        pub fn chat_migration(&self) -> Option<&ChatMigration> {
             match &self.kind {
                 Common(MessageCommon {
                     media_kind: MediaKind::Migration(chat_migration), ..
-                }) => Some(*chat_migration),
+                }) => Some(chat_migration),
                 _ => None,
             }
         }
 
         #[must_use]
-        pub fn migrate_to_chat_id(&self) -> Option<ChatId> {
+        pub fn migrate_to_chat_id(&self) -> Option<&ChatId> {
             match &self.kind {
                 Common(MessageCommon {
                     media_kind: MediaKind::Migration(ChatMigration::To { chat_id }),
                     ..
-                }) => Some(*chat_id),
+                }) => Some(chat_id),
                 _ => None,
             }
         }
 
         #[must_use]
-        pub fn migrate_from_chat_id(&self) -> Option<ChatId> {
+        pub fn migrate_from_chat_id(&self) -> Option<&ChatId> {
```

Comment:
I don't like that we have a `&ChatId`, it's pointless...

Can you at least leave a fixme here?

```rust
// FIXME: remove references to small values (requires changing `define_message_ext`)
```